crates/sui-core/src/authority/authority_store.rs

```diff
@@ -33,9 +33,10 @@ pub type GatewayStore = SuiDataStore<EmptySignInfo>;
 
 pub type InternalSequenceNumber = u64;
 
-type CertLockGuard<'a> = LockGuard<'a>;
+type CertLockGuard = LockGuard;
```

Comment:
Make that a new stuct I suggest: `CertLockGuard(LockGuard)` . Define a `Drop` trait on it to clean up the table after the lock is dropped.

Reply:
Defining a `Drop` may not work because of the usage of tokio::sync::RwLock (which has async lock methods which `Drop` won't support) to protect the table. We also can't really use a blocking lock in this case as it is held across an `await`.  The only other way I can think of to make this work is to spawn a task inside drop and pass it the lock id to trigger a cleanup but this may become expensive soon if we do this for every lock. My subsequent diff which does a periodic cleanup seemed like a good middle ground. Making a new wrapper struct `CertLockGuard` like you suggested however. Happy to address your comment in a future diff if we still want to take a different course of action for lock cleanup.